simcon_project/conversation_templates/views/template_management.py

```diff
@@ -141,8 +154,36 @@ class TemplateDeleteView(BSModalDeleteView):
     success_message = 'Success: Book was deleted.'
     success_url = reverse_lazy('management:main')
 
-
-def remove_template(request, pk):
+    def get(self, request, *args, **kwargs):
+        """
+        Override post to send template name and name of assignment that
+        will be removed as context to the template
+        """
+        super().get(request, *args, **kwargs)
+        this_template = ConversationTemplate.objects.get(pk=self.kwargs['pk'])
+        assignments = this_template.assignments.all()
+        to_delete = []
+        for assignment in assignments:
+            if assignment.conversation_templates.all().count() == 1:
+                to_delete.append(assignment.name)
+        context = {"template_name": this_template.name, "assignments": to_delete}
+        return render(request, self.template_name, context)
+
+    def post(self, request, *args, **kwargs):
+        """
+        Override post to remove assignment if the template being deleted
+        is the only one in the assignment.
+        """
+        this_template = ConversationTemplate.objects.get(pk=self.kwargs['pk'])
+        assignments = this_template.assignments.all()
+        for assignment in assignments:
+            if assignment.conversation_templates.all().count() == 1:
+                assignment.delete()
+        super().post(request, *args, **kwargs)
+        return redirect(reverse('management:main'))
+
+
+def RemoveTemplate(request, pk):
```

Comment:
Functions should be snake-cased. At some point, I think we'll do a sweep of all the view functions to properly case them.

Reply:
I'm getting conflicting info, I used to have it snake case since pycharm tells me to, but I was told to change it to this way in another PR

Reply:
Hmm. I'm going with the Django docs on this one

Reply:
I'll fight whoever says otherwise lol

Reply:
I wish you had told me when I had commented it on all your files Eric! haha